src/client/main.c

```diff
@@ -57,18 +93,31 @@ static int get_opts(int argc, char *argv[], char **opt_filter_glob) {
 
 int main(int argc, char *argv[]) {
         int r = 0;
-
-        _cleanup_free_ char *opt_filter_glob = NULL;
-        r = get_opts(argc, argv, &opt_filter_glob);
+        _cleanup_command_ Command *command = new_command();
+        r = parse_cli_opts(argc, argv, command);
         if (r < 0) {
+                usage();
                 return EXIT_FAILURE;
         }
-        if (no_action) {
+        if (command->op == NULL && command->is_help) {
```

Comment:
Yes, in case `bluechictl --help` is called

Reply:
Yes, but then also `command->is_help` is `True`, right? If yes, then isn't checking `command->op == NULL` redundant?

Reply:
No, because if `command->is_help == true` and `command->op != NULL` we want to proceed to `command_execute()` and call the method-specific usage function.

Reply:
I see. Thanks for the explanation